src/wallet/wallet.cpp

```diff
@@ -417,6 +417,12 @@ bool CWallet::ChangeWalletPassphrase(const SecureString& strOldWalletPassphrase,
 
 void CWallet::SetBestChain(const CBlockLocator& loc)
 {
+    LOCK(cs_wallet); //nWalletMaxVersion
+    unsigned int keypool_min = GetArg("-keypoolmin", DEFAULT_KEYPOOL_MIN);
+    if (IsHDEnabled() && ((CanSupportFeature(FEATURE_HD_SPLIT) && setInternalKeyPool.size() < keypool_min) || (setExternalKeyPool.size() < keypool_min))) {
```

Comment:
In commit "[wallet] Add keypoolmin"

As alluded to in https://github.com/bitcoin/bitcoin/pull/10882#issuecomment-318182682 and https://github.com/bitcoin/bitcoin/pull/10882#issuecomment-318322124, this check is not as safe as it appears, because if keypool gets drained below the minimum, then is refilled later after unlocking, there might be missing transactions which will not get rescanned in the interval of time where the keypool was below the minimum. I would at least add a comment here mentioning this, if you don't want to add an `m_update_best_block` bool to fix it.

By the way, another advantage of adding a `m_update_best_block` bool is that it would let us print a log message when the keypool falls below the minimum (right now there is no indication, and we probably don't want to add a log statement here since it would spam the log).